lib/dynflow/extensions/msgpack.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+require 'msgpack'
+
+module Dynflow
+  module Extensions
+    module MsgPack
+      module Time
+        def to_msgpack(out = '')
+          ::MessagePack.pack(self, out)
+          out
+        end
+      end
+
+      ::Time.include ::Dynflow::Extensions::MsgPack::Time
+      ::MessagePack::DefaultFactory.register_type(0x00, Time, packer: MessagePack::Time::Packer, unpacker: MessagePack::Time::Unpacker)
+
+      begin
+        require 'active_support/time_with_zone'
+        ::ActiveSupport::TimeWithZone.include ::Dynflow::Extensions::MsgPack::Time
+        ::MessagePack::DefaultFactory.register_type(0x01, ActiveSupport::TimeWithZone, packer: MessagePack::Time::Packer, unpacker: MessagePack::Time::Unpacker)
```

Comment:
This is slightly wonky. Once `ActiveSupport::TimeWithZone` instance is serialized and then deserialized, it becomes just `Time`, but still `==` to the original instance. Still better then what we had before

Reply:
Not anymore